Makefile

```diff
@@ -64,45 +67,62 @@ serve:
 		--incremental \
 		--trace
 
+
+# =================== Testing locally from host ==================
+# NOTE that these are for convenience of testing from outside of a container. 
+# You could just as easily enter a fresh container shell and run these as any 
+# code changes on host would be reflected inside the container. e.g. 
+# ```
+# docker-compose run --rm site bash
+# $ make switch-channel CHANNEL=dev
+# $ tool/test.sh --target ${CHANNEL} --check-links --null-safety
+# ```
 CHANNEL ?= stable
 
-# Switch flutter channels inside the running container
-# NOTE this is experimental and compose is mirroring your 
-# local flutter repo via the volume, so this may have 
-# unexpected consequences on your local setup.
+# Convenience command to run inside of a container to 
+# switch the channel of the flutter submodule. 
+# NOTE that changes will be reflected in your local flutter repo 
 # Usage: `make switch-channel CHANNEL=<channel>`
 switch-channel:
-	
-	docker-compose exec site cd flutter && \
+	cd flutter && \
```

Comment:
Out of curiosity, why is `flutter channel ${CHANNEL}` not used here? Are you looking for finer grained access to additional branches (beyond master/dev/beta/stable, I mean)?